### PR TITLE
Extract dev scripts for regenerating the golden files

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

- extract development scripts for regenerating and verifying  the golden files

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/gen/JsonSpecFileGenerator.scala [hunk 1]
```diff
@@ -71,9 +70,7 @@ class JsonSpecFileGenerator extends AnyFunSuite {
       val existedFileContent =
         FileUtils.readFileToString(filePath.toFile, StandardCharsets.UTF_8)
       withClue(s"Check $filename failed. Please regenerate the ranger policy file by running"
-        + "`KYUUBI_UPDATE=1 build/mvn clean test -Pgen-policy"
-        + " -pl :kyuubi-spark-authz_2.12 -Dtest=none"
-        + " -DwildcardSuites=org.apache.kyuubi.plugin.spark.authz.gen.JsonSpecFileGenerator`.") {
+        + "`sh dev/gen/gen_ranger_spec_json.sh`.") {
```
Comment (pan3793) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/gen/JsonSpecFileGenerator.scala:73]: Remove leading `sh `

File: dev/gen/gen_ranger_spec_json.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+KYUUBI_UPDATE=1 \
+build/mvn clean test \
+  -pl :kyuubi-spark-authz_2.12 \
```
Comment (pan3793) [inline dev/gen/gen_ranger_spec_json.sh:21]: it's likely we are going to support scala 2.13 soon, please use dir path to avoid depending on scala versions

Comment (bowenliang123) [reply dev/gen/gen_ranger_spec_json.sh]: No problem, changed to use module's path.

**Review by pan3793 — APPROVED** [review]

**Comment by bowenliang123** [comment]
Thanks, merged to master.

**Pull request closed without merging**